pdfbox/src/main/java/org/apache/pdfbox/contentstream/PDFStreamEngine.java

```diff
@@ -502,6 +505,9 @@ else if (token instanceof Operator)
             {
                 arguments.add((COSBase) token);
             }
+            if (clock.millis() - startTime > 1000) {
```

Comment:
How do you tell the caller of the function that something went wrong? You need a way to pass that information back and then use that to stat it

Reply:
What do you want the caller to do if it timed out? Return directly without passing other pages?